File: src/Shared/Logistics.Shared.Models/Truck/TruckGeolocationDto.cs
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public class TruckGeolocationDto
{
    public required Guid TruckId { get; set; }
    public required Guid TenantId { get; set; }
    public required GeoPoint CurrentLocation { get; set; }
    public Address? CurrentAddress { get; set; }
    public string? TruckNumber { get; set; }
    public string? DriversName { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/truck/truck-geolocation.dto.ts
import {AddressDto} from "../address.dto";
import {GeoPointDto} from "../geo-point.dto";

export interface TruckGeolocationDto {
  truckId: string;
  tenantId?: string;
  currentLocation?: GeoPointDto;
  currentAddress?: AddressDto;
  truckNumber?: string;
  driversName?: string;
}
